### ISSUE TITLE
Why does transformer share same embedding weights within encoder, decoder and logits?

### ISSUE THREAD

**Issue by godsme** [tensorflow/models#5975]
After reading through the transformer code, one thing I don't understand is that, since the model is for translation task from one language to another, the vocabulary of source and target language is different, but the implementation use the same embedding input between encoder and decoder, even the projection of output.  did I misunderstand anything?

**Comment by stikbuf** [comment]
See section 3.4 of https://arxiv.org/abs/1706.03762 .

**Comment by andreaskoepf** [comment]
> See section 3.4 of https://arxiv.org/abs/1706.03762 .

... refers to https://arxiv.org/abs/1608.05859 "Using the Output Embedding to Improve Language Models" which recommends the three-way weight tying.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
